server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/UsersRepositoryAliasingForwarding.java

```diff
@@ -71,43 +74,16 @@ public void setUsersRepository(UsersRepository usersRepository) {
         this.usersRepository = usersRepository;
     }
 
-    /**
-     * Return null when the mail should be GHOSTed, the username string when it
-     * should be changed due to the ignoreUser configuration.
-     * 
-     * @param sender
-     * @param recipient
-     * @param message
-     * @throws MessagingException
-     */
-    public Collection<MailAddress> processMail(MailAddress sender, MailAddress recipient, MimeMessage message) throws MessagingException {
-        if (recipient == null) {
-            throw new IllegalArgumentException("Recipient for mail to be spooled cannot be null.");
-        }
-        if (message == null) {
-            throw new IllegalArgumentException("Mail message to be spooled cannot be null.");
-        }
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Preconditions.checkNotNull(mail);
+        Preconditions.checkNotNull(mail.getMessage());
 
         if (usersRepository instanceof RecipientRewriteTable) {
-            Mappings mappings;
-            try {
-                mappings = ((RecipientRewriteTable) usersRepository).getMappings(recipient.getLocalPart(), recipient.getDomain());
-            } catch (ErrorMappingException e) {
-                String errorBuffer = "A problem as occoured trying to alias and forward user " + recipient + ": " + e.getMessage();
-                throw new MessagingException(errorBuffer);
-            } catch (RecipientRewriteTableException e) {
-                String errorBuffer = "A problem as occoured trying to alias and forward user " + recipient + ": " + e.getMessage();
-                throw new MessagingException(errorBuffer);
-            }
-
-            if (mappings != null) {
-                return handleMappings(mappings, sender, recipient, message);
-            }
+            RecipientRewriteTable virtualTableStore = (RecipientRewriteTable) usersRepository;
```

Comment:
RecipientRewriteTableProcessor processor should be initialized in init() method. Cast of usersRepository should be done in init() too.